features/step_definitions/meta_steps.rb

```diff
@@ -114,3 +114,22 @@
   end
 end
 
+# To exit successfully if output is obtained - mapi
+Given /^I run the steps #{NUMBER} times or exit when co machine-api is degraded:$/ do |num, steps_string|
```

Comment:
my request was to remove this new step and use `I wait(?: up to #{NUMBER} seconds)? for the steps to pass:` instead. See our slack conversation.

Reply:
Hey @akostadinov but , it wont give the same result if I make that change , we will not know if the clusteroperator machine-api , actually degraded ..
Here is what I changed -  And I wait up to 300 seconds for the steps to pass:
    """
    Given a pod becomes ready with labels:
The output , is not the intended one , it just deletes pod and waits the co doesnt get degraded by deleting ones ..
.
.
      [04:35:47] INFO> Shell Commands: rm -r -f -- /root/workdir/3ec7b6b1d17c-
      
      [04:35:47] INFO> Exit Status: 0
      [04:35:47] INFO> === End After Scenario: Machine API Operator should go degraded if any pod controller is crash looping ===

1 scenario (1 passed)
5 steps (5 passed)
0m12.816s
[04:35:47] INFO> === At Exit ===

Reply:
I can't decipher your output. Could you more carefully report what steps you use and what output you get and use code blocks instead of normal text?

Reply:
Hi @akostadinov , here is the complete output - http://pastebin.test.redhat.com/923625 , the scenario executes in less than 13 sec (when I remove the step definition) , whereas it takes around 4 mins once the actual functional test that we want gets executed ..

Reply:
In your custom wait step, you are waiting for some steps to complete and insert a step after them 
```
  step %Q{evaluation of `cluster_operator('machine-api').condition(type: 'Degraded')` is stored in the :co_degraded clipboard}
```

So basically you can remove new step and insert the extra step behind the others like:
```
And I wait up to 180 seconds for the steps to pass:
"""
Given a pod becomes ready with labels:
  | api=clusterapi     |
  | k8s-app=controller |
When admin ensures "#{pod.name}" pod is deleted from the "openshift-machine-api" project
Then the expression should be true> cluster_operator('machine-api').condition(type: 'Degraded')["status"]=="False"
"""
```
What I see in your log is that you have split the steps into separate wait steps which breaks the semantics.

Reply:
Hi @akostadinov  , I might not be able to express clearly , but this is ok for test to have machine-api  degraded as true at first and then false , but we are not aware when it will go degraded for few seconds , so I think it is not breaking semantics .. here is what we are trying to validate - https://bugzilla.redhat.com/show_bug.cgi?id=1859221#c5

Reply:
I pasted wrong step. What I mean is to do
```
And I wait up to 180 seconds for the steps to pass:
"""
Given a pod becomes ready with labels:
  | api=clusterapi     |
  | k8s-app=controller |
When admin ensures "#{pod.name}" pod is deleted from the "openshift-machine-api" project
Then the expression should be true> cluster_operator('machine-api').condition(type: 'Degraded')["status"]=="True"
"""
```

And then, once you know status became degraded, you can continue with the rest of the steps. This is basically what your new step is doing AFAIU. No?

Reply:
you are correct logically , but  , when I use it the way you suggested above it is unable to find pod.name and never kill machine-controller pod 
Checked it in debugging shell :
"[1] pry(#<BushSlicer::DefaultWorld>)> cluster_operator('machine-api').condition(type: 'Degraded')
=> {"lastTransitionTime"=>"2020-12-15T08:57:05Z", "reason"=>"AsExpected", "status"=>"False", "type"=>"Degraded"}
[2] pry(#<BushSlicer::DefaultWorld>)> pod.name
=> "\#{pod.name}""
When I used the way I have : it picks up the podname and keep destroying it until we get degraded status as "True"
Debugging shell when used with meta_step.rb : 
"[1] pry(#<BushSlicer::DefaultWorld>)> pod.name
=> "machine-api-controllers-7f67dc495-mgwt6"
"
Might be some issue with the way I coded it .. will check , but functionally works as expected when I used the way I have used ..

Reply:
@miyadav , please start using code blocks when pasting something. It takes much more effort to understand what you want to say otherwise. I'm not sure why you never do it but it is really important to do.

The issue is that the step `And I wait up to 180 seconds for the steps to pass:` does not process steps individually and doesn't handle resolution of expressions. It is not very easy to fix but is much better to improve the existing step instead of creating a new step with limited application.

Until somebody has time to improve the step, I'd suggest you to use step
```
/^I remove all #{WORD}(?: in the#{OPT_QUOTED} project) with labels:$/
```
i.e.
```
Then I remove all pods with labels:
  | api=clusterapi     |
  | k8s-app=controller |
```
Will this do?

Reply:
thanks @akostadinov  for pointing  out , I will use code blocks . 
I am not sure , above will help , as it will still not monitor the cluster operator status  ?

Reply:
@miyadav , what you are presently doing is delete the pods and exit when status becomes degraded. Instead of writing a new step, you can use the existing wait step and do a last check
`Then the expression should be true> cluster_operator('machine-api').condition(type: 'Degraded')["status"]=="True"` as I suggested above. What other monitoring is needed? I can't understand what is the issue.

```
And I wait up to 180 seconds for the steps to pass:
"""
Then I remove all pods with labels:
  | api=clusterapi     |
  | k8s-app=controller |
Then the expression should be true> cluster_operator('machine-api').condition(type: 'Degraded')["status"]=="True"
```

Or something similar. It is almost the same of what you do, except that you are waiting for pod to become available. Accoirding to your description it doesn't sound to me that pod status checking is important. Or what else is missing?